app/Controllers/PrestitiController.php

```
@@ -755,8 +785,15 @@ public function update(Request $request, Response $response, mysqli $db, int $id
         // giornata singola) è lecita: createReservation accetta end == start,
         // quindi un rifiuto strettamente esclusivo renderebbe immodificabili
         // i prestiti a giornata nati dal calendario utente.
-        if (strtotime($newScadenza) === false || strtotime($newPrestito) === false
-            || strtotime($newScadenza) < strtotime($newPrestito)) {
+        // F028 (#335) + strict-ISO (#337): valida entrambe le date in Y-m-d
+        // STRETTO senza strtotime, con CODICI D'ERRORE SEPARATI — formato non
+        // valido vs range invertito — così l'admin riceve un messaggio azionabile.
+        // DateHelper rifiuta date ambigue o inesistenti (2026-02-30) e byte NUL;
+        // la ri-validazione sotto lock più in basso applica lo stesso criterio.
+        if (!\App\Support\DateHelper::isISODateFormat($newPrestito) || !\App\Support\DateHelper::isISODateFormat($newScadenza)) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Support\DateHelper' in method 'update'. (undefined)

(StaticAccess)